gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/DistributedIdentifyContents.java

```diff
@@ -0,0 +1,306 @@
+          | NoSuchMethodException e) {
+        throw new IllegalArgumentException(
+            String.format("Could not initialize '%s': ", clientBuilderClassName), e);
+      }
+    }
+    return (NessieApiV1) builder.fromConfig(configuration::get).build(NessieApiV1.class);
+  }
+
+  /**
+   * Compute the bloom filter per content id by walking all the live references in a distributed way
+   * using spark.
+   *
+   * @param references list of live references to walk.
+   * @param cutOffTimestampPerRef map of cutoff timestamp per reference.
+   * @param defaultCutOffTimestamp default cutoff timestamp to be used.
+   * @return map of {@link ContentBloomFilter} per content-id.
+   */
+  public Map<String, ContentBloomFilter> getLiveContentsBloomFilters(
+      List<Reference> references,
+      Map<String, Instant> cutOffTimestampPerRef,
+      Instant defaultCutOffTimestamp) {
+    List<Map<String, ContentBloomFilter>> bloomFilterMaps =
+        new JavaSparkContext(session.sparkContext())
+            .parallelize(references, getTaskCount(references))
+            .map(
+                reference ->
+                    computeLiveContents(
+                        cutOffTimestampPerRef.getOrDefault(
+                            reference.getName(), defaultCutOffTimestamp),
+                        reference))
+            .collect();
+    return mergeLiveContentResults(bloomFilterMaps);
+  }
+
+  /**
+   * Gets the expired contents per content id by walking all the live and dead references in a
+   * distributed way using spark and checking the contents against the live bloom filter results.
+   *
+   * @param liveContentsBloomFilterMap live contents bloom filter per content id.
+   * @param references list of all the references to walk (live and dead)
+   * @return {@link IdentifiedResult} object.
+   */
+  public IdentifiedResult getIdentifiedResults(
+      Map<String, ContentBloomFilter> liveContentsBloomFilterMap, List<Reference> references) {
+
+    List<IdentifiedResult> results =
+        new JavaSparkContext(session.sparkContext())
+            .parallelize(references, getTaskCount(references))
+            .map(reference -> computeExpiredContents(liveContentsBloomFilterMap, reference))
+            .collect();
+    IdentifiedResult identifiedResult = new IdentifiedResult();
+    results.forEach(result -> identifiedResult.addNewContentValues(result.getContentValues()));
+    return identifiedResult;
+  }
+
+  protected Map<String, ContentBloomFilter> computeLiveContents(
+      Instant cutOffTimestamp, Reference reference) {
+    try (NessieApiV1 api = getApi(config)) {
+      return walkLiveCommitsInReference(
+          api,
+          reference,
+          commitMeta ->
+              // If the commit time is newer than (think: greater than or equal to) cutoff-time,
+              // then commit is live.
+              Objects.requireNonNull(commitMeta.getCommitTime()).compareTo(cutOffTimestamp) >= 0);
```

Comment:
Nit: The usage of `Objects.requireNonNull` here adds no value. Either add a descriptiive message or just omit the `Objects.requireNonNull`

Same for the user usages of `Objects.requireNonNull` below

Reply:
I added for IDE warning. I will remove it if not required.